sematic/utils/exceptions.py

```diff
@@ -46,8 +106,10 @@ def format_exception_for_run(
     else:
         repr_ = traceback.format_exc()
 
+    assert isinstance(cause_exception, BaseException)
```

Comment:
Should this really fail? I imagine this would completely swallow the initial weird error / illegal state, right when you would want as much information as possible.
How about instead we just log an error and fall back to BaseException as the ancestor (or leave empty)?